provider/core/src/resource.rs

```diff
@@ -5,128 +5,210 @@
 //! Resource paths and related types.
 
 use alloc::borrow::Cow;
-use alloc::format;
-use alloc::string::String;
 use alloc::string::ToString;
 
 use crate::error::{DataError, DataErrorKind};
-use core::borrow::Borrow;
+use crate::helpers;
 use core::default::Default;
 use core::fmt;
 use core::fmt::Write;
 use icu_locid::LanguageIdentifier;
-use tinystr::{TinyStr16, TinyStr4};
 use writeable::{LengthHint, Writeable};
 
-/// A top-level collection of related resource keys.
-#[non_exhaustive]
-#[derive(PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Debug)]
-pub enum ResourceCategory {
-    Core,
-    Calendar,
-    DateTime,
-    Decimal,
-    LocaleCanonicalizer,
-    Plurals,
-    TimeZone,
-    Properties,
-    ListFormatter,
-    Segmenter,
-    PrivateUse(TinyStr4),
-}
+/// A compact hash of a [`ResourceKey`]. Useful for keys in maps.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Hash)]
+#[repr(transparent)]
```

Comment:
We don't need the `repr`s here

Reply:
I'd like to keep the `repr(transparent)` on ResourceKeyHash because that one will be used as a ULE in the BlobProvider.  But I removed the `repr(C)` on ResourceKey.